riscv-page-tables/src/page_table.rs

```diff
@@ -977,6 +988,26 @@ impl<T: PagingMode> GuestStagePageTable<T> {
         Ok(mapper)
     }
 
+    fn check_and_increment_vaddr(
+        mut va: PageAddr<T::MappedAddressSpace>,
+        end: PageAddr<T::MappedAddressSpace>,
+        page_size: PageSize,
+    ) -> Result<Option<PageAddr<T::MappedAddressSpace>>> {
```

Comment:
nit: i find `Result<Option<>` to be an awkward return value. instead of returning `Ok(None)` in the `va == end` case, could you instead have the outer loop in which this called go until `va == end`?

Reply:
I agree, this is much simpler and much nicer :)